### PR TITLE
Migrated account to account_logins to request helpers

### PR DESCRIPTION
**Pull request by ErikZilber**
## 📝 Description

Migrated the following files from directly referencing the Resty client in favor of using the new request helpers:
account.go
account_availability.go
account_betas.go
account_events.go
account_invoices.go
account_logins.go

### CODE REVIEW

File: account_betas.go [hunk 1]
```diff
@@ -102,13 +80,11 @@ func (c *Client) JoinBetaProgram(ctx context.Context, opts AccountBetaProgramCre
 	if err != nil {
 		return nil, err
 	}
-
 	e := "account/betas"
-	req := c.R(ctx).SetResult(&AccountBetaProgram{}).SetBody(string(body))
-	r, err := coupleAPIErrors(req.Post(e))
+	response, err := doPOSTRequest[AccountBetaProgram](ctx, c, e, body)
```
Comment (lgarber-akamai) [inline account_betas.go:84]: The new `doPOSTRequest` and `doPUTRequest` helpers accept instances of structs and handle all of the serialization behind the scenes, so we should be able to pass in `opts` and remove the JSON marshaling logic above 🙂 

```suggestion
	 response, err := doPOSTRequest[AccountBetaProgram](ctx, c, e, opts)
```

File: account_availability.go [hunk 2]
```diff
@@ -20,46 +18,24 @@ type AccountAvailability struct {
...
-	return response.Data, nil
+
+	return response, nil
 }
 
 // GetAccountAvailability gets the resources availability in a region to the customer.
 func (c *Client) GetAccountAvailability(ctx context.Context, regionID string) (*AccountAvailability, error) {
-	req := c.R(ctx).SetResult(&AccountAvailability{})
 	regionID = url.PathEscape(regionID)
 	b := fmt.Sprintf("account/availability/%s", regionID)
```
Comment (lgarber-akamai) [inline account_availability.go:34]: We recently added a new [formatAPIPath](https://github.com/linode/linodego/blob/main/request_helpers.go#L198) helper to automatically handle escaping path segments:

```suggestion
	b := formatAPIPath("account/availability/%s", regionID)
```

Comment (ErikZilber) [reply account_availability.go:34]: Thanks! I'll fix this across all of the files in the PR.

**Review by lgarber-akamai — APPROVED** [review]
Looks great!

**Review by zliang-akamai — APPROVED** [review]
Things look much simpler and cleaner now, nice job!

**Pull request merged**